server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/HttpJmapAuthentication.java

```
@@ -21,32 +21,54 @@
 
+    private static final ConditionFactory CALMLY_AWAIT = Awaitility.with()
+            .pollInterval(Duration.FIVE_HUNDRED_MILLISECONDS)
+            .and().with()
+            .pollDelay(Duration.FIVE_HUNDRED_MILLISECONDS)
+            .await();
+
     public static AccessToken authenticateJamesUser(URIBuilder uriBuilder, String username, String password) throws ClientProtocolException, IOException, URISyntaxException {
         String continuationToken = getContinuationToken(uriBuilder, username);
 
-        Response response = Request.Post(uriBuilder.setPath("/authentication").build())
-                .bodyString("{\"token\": \"" + continuationToken + "\", \"method\": \"password\", \"password\": \"" + password + "\"}", 
-                        ContentType.APPLICATION_JSON)
-                .setHeader("Accept", ContentType.APPLICATION_JSON.getMimeType())
-                .execute();
-        
+        Optional<Response> response = CALMLY_AWAIT.atMost(30, TimeUnit.SECONDS)
+        .until(() -> postAuthenticate(uriBuilder, password, continuationToken), new IsPresentMatcher());
+
         return AccessToken.fromString(
-                    JsonPath.parse(response.returnContent().asString())
+                    JsonPath.parse(response.get().returnContent().asString())
                     .read("accessToken"));
     }
 
+    private static Optional<Response> postAuthenticate(URIBuilder uriBuilder, String password, String continuationToken) {
+        try {
+            return Optional.of(Request.Post(uriBuilder.setPath("/authentication").build())
+                    .bodyString("{\"token\": \"" + continuationToken + "\", \"method\": \"password\", \"password\": \"" + password + "\"}", 
+                            ContentType.APPLICATION_JSON)
+                    .setHeader("Accept", ContentType.APPLICATION_JSON.getMimeType())
+                    .execute());
+        } catch (IOException | URISyntaxException e) {
+            return Optional.empty();
```

you can tell him to ignore exceptions, but that's in 1.6.5 instead of 1.6.3 :)